jax/lax.py

```diff
@@ -1755,17 +1755,30 @@ def _select_transpose_rule(t, pred, on_true, on_false):
             select(pred, zeros, t) if on_false is None else None]
 
 def _select_batch_rule(batched_args, batch_dims, **unused_kwargs):
```

Comment:
(I haven't studied this change carefully yet.)

Would this rule be simplified if we generalized the semantics of `lax.select` to support more general broadcasting? 

The XLA `Select` semantics are a little awkward (e.g., only scalar broadcasting) so I wrote a wrapper that does general broadcasting, not just scalar/non-scalar:
https://github.com/google/jax/blob/master/jax/lax.py#L1001

Reply:
Yes, definitely. That's great to know about! These commits could probably be simplified by using that helper function.